### PR TITLE
fix: make location configurable

### PR DESCRIPTION
**Pull request by jaulz**
This PR will add full SSR support.

### CODE REVIEW

**Comment by jaulz** [comment]
@bakerkretzmar this is my suggestion for https://github.com/tighten/ziggy/issues/431 😊 It would be great if you could have a look!
@aarondfrancis maybe also interesting for you 😊

**Comment by aarondfrancis** [comment]
Love it, this would do the trick!

**Comment by aarondfrancis** [comment]
Here's a workaround for now: https://aaronfrancis.com.wip/2021/using-ziggy-with-inertia-server-side-rendering

File: src/js/Router.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default class Router extends String {
         super();
 
         this._config = config ?? Ziggy ?? globalThis?.Ziggy;
-        this._config = { ...this._config, absolute };
+        this._config = { location: {}, ...this._config, absolute };
```
Comment (bakerkretzmar) [inline src/js/Router.js:18]: This isn't strictly necessary, right? As long as we check for null/undefined when we access it below with something like `this._config.location?.host`?

**Comment by aarondfrancis** [comment]
> Thanks a ton for this PR @jaulz, and for your post @aarondfrancis! This looks great to me, I'll wait for you to take a look at my questions and I'm just going to push up some formatting tweaks and then it should be good to go.
> 
> Am I right that after this PR, using Ziggy with Inertia and SSR would look something like this?
> 
> ```js
> import route from 'ziggy';
> 
> export default async function (event) {
>     return await createInertiaApp({
>         // ssr stuff...
>         setup({app, props, plugin}) {
> 
>             const Ziggy = { ...event.props.ziggy, location: new URL(event.props.ziggy.url) };
> 
>             return createSSRApp({
>                 render: () => h(app, props),
>             }).mixin({
>               methods: {
>                   route: (name, params, absolute, config = Ziggy) => route(name, params, absolute, config),
>               },
>           })
>         },
>     });
> }
> ```

Yes, I think so! And it's beautiful 😍 

Thanks @bakerkretzmar!

**Comment by jaulz** [comment]
@bakerkretzmar thanks for cross checking and I incorporated your feedback 😊

**Comment by jaulz** [comment]
@bakerkretzmar do you think we could merge it in this state? Otherwise I am happy to rework it.

**Comment by bakerkretzmar** [comment]
Thanks a ton @jaulz!

**Comment by yashiroiori** [comment]
> Thanks a ton for this PR @jaulz, and for your post @aarondfrancis! This looks great to me, I'll wait for you to take a look at my questions and I'm just going to push up some formatting tweaks and then it should be good to go.
> 
> Am I right that after this PR, using Ziggy with Inertia and SSR would look something like this?
> 
> ```js
> import route from 'ziggy';
> 
> export default async function (event) {
>     return await createInertiaApp({
>         // ssr stuff...
>         setup({app, props, plugin}) {
> 
>             const Ziggy = { ...event.props.ziggy, location: new URL(event.props.ziggy.url) };
> 
>             return createSSRApp({
>                 render: () => h(app, props),
>             }).mixin({
>               methods: {
>                   route: (name, params, absolute, config = Ziggy) => route(name, params, absolute, config),
>               },
>           })
>         },
>     });
> }
> ```

From where import "createSSRApp", I'm using inertiajs with react

**Comment by bakerkretzmar** [comment]
I think SSR is still in early access for Inertia sponsors only: https://inertiajs.com/server-side-rendering

**Comment by yashiroiori** [comment]
> I think SSR is still in early access for Inertia sponsors only: https://inertiajs.com/server-side-rendering

Yes, but in examples are in this page from where import createSSRApp??

**Comment by bakerkretzmar** [comment]
I would guess something like `import { createSSRApp } from '@inertiajs/inertia-vue3'` but I'm not in early access so I'm not sure 😅

**Comment by heychazza** [comment]
Hey guys,

> Thanks a ton for this PR @jaulz, and for your post @aarondfrancis! This looks great to me, I'll wait for you to take a look at my questions and I'm just going to push up some formatting tweaks and then it should be good to go.
> 
> Am I right that after this PR, using Ziggy with Inertia and SSR would look something like this?
> 
> ```js
> import route from 'ziggy';
> 
> export default async function (event) {
>     return await createInertiaApp({
>         // ssr stuff...
>         setup({app, props, plugin}) {
> 
>             const Ziggy = { ...event.props.ziggy, location: new URL(event.props.ziggy.url) };
> 
>             return createSSRApp({
>                 render: () => h(app, props),
>             }).mixin({
>               methods: {
>                   route: (name, params, absolute, config = Ziggy) => route(name, params, absolute, config),
>               },
>           })
>         },
>     });
> }
> ```

trying this, I am told event doesn't exist, has anything changed since for configuring SSR?

**Comment by anburocky3** [comment]
What is the equalent of using React?

**Pull request merged**